libs/@local/hql/span/src/storage.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{tree::SpanNode, Span, SpanId};
+
+/// A collection of spans within a single source file.
+///
+/// This struct is used to store information about multiple spans within a single source file.
+pub struct SpanStorage<S> {
+    arena: Vec<S>,
```

Comment:
no, it was a `la_arena::Arena` before^^, but I changed it halfway through to a simple vec (`la_arena` is no different - but used internally in RA), chose against using it because it doesn't allow indexing with `get`, and it's an extra dependency we don't really need.